ticdc: add new doc for TiCDC data replication capabilities (#20967)
pingcap/docs: ticdc/ticdc-data-replication-capabilities.md

---

> title: Analysis of TiCDC's Data Replication Capabilities
> summary: Learn TiCDC's data Replication capabilities.
> ---
> 
> # Analysis of TiCDC's Data Replication Capabilities

Reviewer: ```suggestion
title: TiCDC Data Replication Capabilities
summary: Learn the data replication capabilities of TiCDC.
---

# TiCDC Data Replication Capabilities
```

+ TiCDC guarantees eventual consistency of transactions. [redo log](/ticdc/ticdc-sink-to-mysql.md#eventually-consistent-replication-in-disaster-scenarios) provides the final consistency guarantee in disaster recovery scenarios. [Syncpoint](/ticdc/ticdc-upstream-downstream-check.md#enable-syncpoint) provides consistent snapshot reads and data consistency checks.

## Downstream Systems Supported by TiCDC

TiCDC supports replication data to multiple downstreams, including:

- [TiDB and MySQL-compatible databases](/ticdc/ticdc-sink-to-mysql.md)
- [Message Queue(MQ) Sink](/ticdc/ticdc-changefeed-config.md#sink), such as [Kafka](/ticdc/ticdc-sink-to-kafka.md), [Pulsar](/ticdc/ticdc-sink-to-pulsar.md)
- [Cloud Storage Sink](/ticdc/ticdc-sink-to-cloud-storage.md), such as Amazon S3, GCS, Azure Blob Storage, and NFS.
- [Integrate Data with Confluent Cloud and Snowflake](/ticdc/integrate-confluent-using-ticdc.md)
- [Integrate Data with Apache Kafka and Apache Flink](/ticdc/replicate-data-to-kafka-and-flink.md)

# Data Replication Scope of TiCDC

TiCDC supports the following upstream data changes:

> - Support
>   - DDL and DML statements(exclude system tables).
>   - Index Action(`ADD INDEX`, `CREATE INDEX`): to reduce the impact on the replication delay of Changefeed, if the downstream is TiDB, TiCDC will [asynchronously execute the DDL operations of creating and adding indexes](/ticdc/ticdc-ddl.md#asynchronous-execution-of-add-index-and-create-index-ddls)
>   - Foreign Key Constraints DDL(`ADD FOREIGN KEY`): TiCDC is not responsible for synchronizing the settings of upstream system variables, such as [foreign_key_checks](/system-variables.md#foreign_key_checks). Therefore, you need to set appropriate system variables in the downstream to determine whether the downstream foreign key constraint check is enabled.

Reviewer: ```suggestion
+ **Supported:**

    - DDL and DML statements(excluding system tables).
    - Index operations (`ADD INDEX`, `CREATE INDEX`): to reduce the impact on changefeed replication latency, if the downstream is TiDB, TiCDC [asynchronously executes the `ADD INDEX` and `CREATE INDEX` DDL operations](/ticdc/ticdc-ddl.md#asynchronous-execution-of-add-index-and-create-index-ddls).
    - Foreign key constraint DDL statements (`ADD FOREIGN KEY`): TiCDC does **not** replicate upstream system variable settings. You need to manually configure [`foreign_key_checks`](/system-variables.md#foreign_key_checks) in the downstream to determine whether the downstream foreign key constraint check is enabled.
```

> - Not Support:
>   - DDL and DML statements executed in upstream system tables (including `mysql.*` and `information_schema.*`).
>   - DDL and DML statements executed in upstream temporary tables.
>   - DQL (Data Query Language) and DCL (Data Control Language) statements.

Reviewer: ```suggestion
- **Not supported**:
    - DDL and DML statements executed in upstream system tables (including `mysql.*` and `information_schema.*`).
    - DDL and DML statements executed in upstream temporary tables.
    - DQL (Data Query Language) and DCL (Data Control Language) statements.
```

> ## TiCDC Usage Limitations​

Reviewer: ```suggestion
## Limitations​
```

> - TiCDC does not support some scenarios. For details, see [Unsupported scenarios](/ticdc/ticdc-overview.md#unsupported-scenarios).
> - TiCDC only checks the integrity of the upstream changes received internally, and does not participate in checking whether the data changes meet the various downstream constraints. If a data change that does not meet the downstream constraints is encountered, TiCDC will report an error when writing to the downstream.

Reviewer: ```suggestion
- TiCDC does not support certain scenarios. For details, see [unsupported scenarios](/ticdc/ticdc-overview.md#unsupported-scenarios).
- TiCDC only verifies the integrity of upstream data changes. It does not validate whether the changes conform to downstream constraints. If the data violates downstream constraints, TiCDC will return an error when writing to the downstream.
```

- TiCDC ensures eventual consistency of transactions. With [redo log](/ticdc/ticdc-sink-to-mysql.md#eventually-consistent-replication-in-disaster-scenarios) enabled, TiCDC can guarantee eventual consistency in disaster recovery scenarios. With [Syncpoint](/ticdc/ticdc-upstream-downstream-check.md#enable-syncpoint) enabled, TiCDC supports consistent snapshot reads and data consistency validation.

## Supported downstream systems

TiCDC supports replicating data to various downstream systems, including the following:

- [TiDB database or other MySQL-compatible databases](/ticdc/ticdc-sink-to-mysql.md)
- [Apache Kafka](/ticdc/ticdc-sink-to-kafka.md)
- [Message Queue (MQ)-type sinks](/ticdc/ticdc-changefeed-config.md#sink), such as [Pulsar](/ticdc/ticdc-sink-to-pulsar.md)
- [Storage services (Amazon S3, GCS, Azure Blob Storage, and NFS)](/ticdc/ticdc-sink-to-cloud-storage.md)
- [Snowflake, ksqlDB, SQL Server via Confluent Cloud integration](/ticdc/integrate-confluent-using-ticdc.md)

> - [Apache Flink for consuming Kafka-replicated data](/ticdc/replicate-data-to-kafka.md)

Reviewer: ```suggestion
- [Apache Flink for consuming Kafka-replicated data](/replicate-data-to-kafka.md)
```

> - [Apache Flink for consuming Kafka-replicated data](/ticdc/replicate-data-to-kafka-and-flink.md)

Reviewer: ```suggestion
- [Apache Flink for consuming Kafka-replicated data](/ticdc/replicate-data-to-kafka.md)
```

## Scope of data replication

TiCDC supports the following types of upstream data changes:

+ **Supported:**

    - DDL and DML statements(excluding system tables).
    - Index operations (`ADD INDEX`, `CREATE INDEX`): to reduce the impact on changefeed replication latency, if the downstream is TiDB, TiCDC [asynchronously executes the `ADD INDEX` and `CREATE INDEX` DDL operations](/ticdc/ticdc-ddl.md#asynchronous-execution-of-add-index-and-create-index-ddls).

>     - Foreign key constraint DDL statements (`ADD FOREIGN KEY`): TiCDC does **not** replicate upstream system variable settings. You need to manually configure [`foreign_key_checks`](/system-variables.md#foreign_key_checks) in the downstream to determine whether the downstream foreign key constraint check is enabled. TiCDC does **not** validate whether the data satisfies foreign key constraints.

Reviewer: ```suggestion
    - Foreign key constraint DDL statements (`ADD FOREIGN KEY`): TiCDC does **not** replicate upstream system variable settings. You need to manually configure [`foreign_key_checks`](/system-variables.md#foreign_key_checks) in the downstream to determine whether the downstream foreign key constraint check is enabled. TiCDC does **not** validate whether the data complies with foreign key constraints.
```

+ **Not supported**:

    - DDL and DML statements executed in upstream system tables (including `mysql.*` and `information_schema.*`).
    - DDL and DML statements executed in upstream temporary tables.
    - DQL (Data Query Language) and DCL (Data Control Language) statements.

## Limitations​

- TiCDC does not support certain scenarios. For details, see [unsupported scenarios](/ticdc/ticdc-overview.md#unsupported-scenarios).

> - TiCDC only verifies the integrity of upstream data changes. It does not validate whether the changes conform to downstream constraints. If the data violates downstream constraints, TiCDC will return an error when writing to the downstream.

Reviewer 2: ```suggestion
- TiCDC only verifies the integrity of upstream data changes. It does not validate whether the changes conform to upstream or downstream constraints. If the data violates downstream constraints, TiCDC will return an error when writing to the downstream. For instance, TiCDC does **not** do foreign key validation at all.
```

> - TiCDC only verifies the integrity of upstream data changes. It does not validate whether the changes conform to upstream or downstream constraints. If the data violates downstream constraints, TiCDC will return an error when writing to the downstream. For instance, TiCDC does **not** do foreign key validation at all.

Reviewer: ```suggestion
- TiCDC only verifies the integrity of upstream data changes. It does not validate whether the changes conform to upstream or downstream constraints. If the data violates downstream constraints, TiCDC will return an error when writing to the downstream. For example, TiCDC does **not** perform any foreign key validation.
```
